src/Controls/src/Core/Handlers/Items/Android/Adapters/ReorderableItemsViewAdapter.cs

```diff
@@ -10,9 +10,12 @@ public class ReorderableItemsViewAdapter<TItemsView, TItemsViewSource> : Groupab
 		where TItemsView : ReorderableItemsView
 		where TItemsViewSource : IGroupableItemsViewSource
 	{
+		private readonly bool _hasHeader;
+
 		public ReorderableItemsViewAdapter(TItemsView reorderableItemsView,
 			Func<View, Context, ItemContentView> createView = null) : base(reorderableItemsView, createView)
 		{
+			_hasHeader = reorderableItemsView.Header is not null;
```

Comment:
does this work if it's HeaderTemplate ?

Reply:
I apologize for keeping you waiting.
Yes it works.
I tested it like this

```xml
<ContentPage
    x:Class="Maui.Controls.Sample.Pages.Controls.CollectionViewGalleries.ReorderingGalleries.HeaderTemplateExample"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml">
    <CollectionView
        CanReorderItems="True"
        Header="{Binding .}"
        ItemsSource="{Binding Items}">
        <CollectionView.HeaderTemplate>
            <DataTemplate>
                <Label Text="{Binding Title}" />
            </DataTemplate>
        </CollectionView.HeaderTemplate>
        <CollectionView.ItemTemplate>
            <DataTemplate>
                <StackLayout Padding="16">
                    <Label Text="{Binding Name}" />
                    <Label Text="{Binding Age}" />
                </StackLayout>
            </DataTemplate>
        </CollectionView.ItemTemplate>
    </CollectionView>
</ContentPage>
```

```C#
namespace Maui.Controls.Sample.Pages.Controls.CollectionViewGalleries.ReorderingGalleries
{
    [XamlCompilation(XamlCompilationOptions.Compile)]
    public partial class HeaderTemplateExample : ContentPage
    {
        public HeaderTemplateExample()
        {
            InitializeComponent();
	    BindingContext = new ReorderingHeaderTemplateViewModel();
        }

        public class ReorderingHeaderTemplateViewModel
        {
            public ReorderingHeaderTemplateViewModel()
            {
                FillCollection();
            }

            public string Title => "It's a title";
            public ObservableCollection<TestItem> Items { get; } = new();

            private void FillCollection()
            {
                for (var i = 0; i < 30; ++i)
                    Items.Add(new TestItem() { Name = $"Item {i}", Age = i });
            }
        }
    }

    public class TestItem
    {
        public string Name { get; set; }
        public int Age { get; set; }
    }
}
```
and it works great.
`_hasHeader is true` when we use HeaderTemplate